src/hamiltonian_type.jl

```diff
@@ -1,5 +1,12 @@
 # classification for different kinds of hamiltonian.
 # algorithm designer will need these type to adopt/reject hamiltonian.
+export AbstractHamiltonian
+export AbstractFermionicHamiltonian
+export AbstractSpinlessFermionicHamiltonian
+export AbstractSpinfulFermionicHamiltonian
+export AbstractElectronHamiltonian
+export AbstractSpinHamiltonian
+export AbstractBosonicHamiltonian
```

Comment:
There will be tooooo many types of hamiltonian in EasyHamiltonian, maybe we would better to be conservative to export them?